server/core/hot_region_storage.go

```diff
@@ -254,16 +254,16 @@ func (h *HotRegionStorage) packHistoryHotRegions(historyHotRegions []HistoryHotR
 	for i := range historyHotRegions {
 		region := &metapb.Region{
 			Id:             historyHotRegions[i].RegionID,
-			StartKey:       historyHotRegions[i].StartKey,
-			EndKey:         historyHotRegions[i].EndKey,
+			StartKey:       HexRegionKey([]byte(historyHotRegions[i].StartKey)),
```

Comment:
I think we can store origin format, just covert it in API.

Reply:
If we use Json.Marshal to Encode the StartKey and EndKey and then use Json.UnMarshal to decode them, the values we get will change. The []byte will be escaped. For example, the original key is "t\x80\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xff\x15_i\x80\x00\x00\x00\x00\x00\x00\x00\xff\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xfa", and after json.Marshal and then UnMarshal becomes "t�\x00\x00\x00\x00\x00\x00\x00\x00\x00��\x15_i�\x00\x00\x00\x00\x00\x00\x00\x00��\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"